lib/types/rules.d.ts

```diff
@@ -2850,6 +2850,22 @@ export interface ESLintRules extends Linter.RulesRecord {
 				 * @default false
 				 */
 				detectObjects: boolean;
+				/**
+				 * @default false
+				 */
+				ignoreEnums?: boolean;
+				/**
+				 * @default false
+				 */
+				ignoreNumericLiteralTypes?: boolean;
+				/**
+				 * @default false
+				 */
+				ignoreReadonlyClassProperties?: boolean;
+				/**
+				 * @default false
+				 */
+				ignoreTypeIndexes?: boolean;
```

Comment:
```suggestion
				ignoreEnums: boolean;
				/**
				 * @default false
				 */
				ignoreNumericLiteralTypes: boolean;
				/**
				 * @default false
				 */
				ignoreReadonlyClassProperties: boolean;
				/**
				 * @default false
				 */
				ignoreTypeIndexes: boolean;
```

For consistency with other options, since we're already using Partial<>.